site_scons/site_tools/roc/config.py

```diff
@@ -290,6 +295,67 @@ def FindConfigGuess(context):
     context.Result('not found')
     return False
 
+def FindPkgConfigPath(context):
+    env = context.env
+
+    context.Message("Searching PKG_CONFIG_PATH...")
+
+    if env.HasArg('PKG_CONFIG_PATH'):
+        context.Result(env['PKG_CONFIG_PATH'])
+        return True
+
+    if env.HasArg('PKG_CONFIG'):
+        pkg_config = env['PKG_CONFIG']
+    elif env.Which('pkg-config'):
+        pkg_config = 'pkg-config'
```

Comment:
It seems this branch (selecting system-wide `pkg-config` tool from PATH) is valid only if we're *not* cross-compiling. Because if we're, out sysroot likely has its own pkg-config directory.

So to pass `crosscompile` variable to this method:

```
conf.FindPkgConfigPath(crosscompile)
```

and if it's set to True, we should skip this `elif` branch.

Other branches, e.g. `env.HasArg('PKG_CONFIG')` or using `libdir/pkgconfig`, seems to be valid for cross-compiling.

Reply:
I also added `crosscompile` check in `AddPkgConfigDependency`, let me know if it is desirable.